src/elements.rs

```diff
@@ -94,28 +93,18 @@ impl ImageBuilder<'_> {
     /// Sets the image data.
     /// The data must be created using [`Clay::data`].
     #[inline]
-    pub fn data(&mut self, data: DataRef) -> &mut Self {
-        self.parent.inner.image.imageData = data.ptr as *mut c_void;
+    pub fn data(&mut self, data: &'render mut ImageElementData) -> &mut Self {
```

Comment:
This doesn't have to be mutable

Reply:
it's because it was already mutable so I just kept it

Reply:
This has to be changed be non-mutable because this will just cause more head ace for users.

Reply:
I agree